### DISCUSSION TITLE
Setting next activity to a previously executed activity programmatically.

### DISCUSSION THREAD

**Discussion by eldhosekpaul18** [elsa-workflows/elsa-core#1288] [Q&A]
Is there a way to set the workflow which is blocked in a user task activity to discard the current blocking activity and start from a previously executed activity?

![image](https://user-images.githubusercontent.com/52571606/103725673-23bf6c80-4ffd-11eb-8069-e64cdf151af8.png)

For example, In the above image, I need to set a workflow instance that is been blocked at Accounts Approval Activity (Waiting for approval or reject user action) to any of the previously executed activity (Admin or Manager Approval) programmatically without triggering a user action at Accounts Approval Activity. The workflow instance needs to stop waiting for Accounts approval user action(Clear the accounts approval blocking activity) and starts from newly set activity.

**Comment by sfmskywalker** [comment]
Although there's no official API exposing such an operation, you can easily do it by manipulating the workflow instance. All you would have to do is update `WorkflowInstance.BlockingActivities` by removing the entry containing the **Accounts Approval** activity reference and add a new entry with the **Admin Approval** or **Manager Approval** activity type and ID.

For example:

```csharp
var workflowInstance = _workflowInstanceStore.GetByIdAsync(...);
workflowInstance.BlockingActivities.Clear();
var adminApprovalActivityId = workflowInstance.Activities.Values.First(x => x.ActivityType == "AdminApproval").Id;
workflowInstance.BlockingActivities.Add(new BlockingActivity { ActivityType = "AdminApproval", ActivityId = adminApprovalActivityId });
```
